src/server/Adaptive.ReactiveTrader.Server.Pricing/PriceSourceAdapters/AdapterBase.cs

```diff
@@ -0,0 +1,120 @@
+using HtmlAgilityPack;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Adaptive.ReactiveTrader.Server.Pricing.PriceSourceAdapters
+{
+  public abstract class AdapterBase
```

Comment:
This feels like it would be better done via composition rather than inheritance.
And I think we can split things up a bit more as well, especially where we have things for parsing HTML documents, which are clearly never going to be used by API.
This looks like it's a bucket of things that are useful across various adapters rather than being a true base class.

adapters.

Reply:
done